workflow/controller/controller.go

```diff
@@ -857,7 +921,13 @@ func (wfc *WorkflowController) addWorkflowInformerHandlers(ctx context.Context)
 					log.Warnf("Workflow FilterFunc: '%v' is not an unstructured", obj)
 					return false
 				}
-				return reconciliationNeeded(un)
+				needed := reconciliationNeeded(un)
+				if !needed {
+					key, _ := cache.MetaNamespaceKeyFunc(un)
+					wfc.recordCompletedWorkflow(key)
+					wfc.throttler.Remove(key)
+				}
```

Comment:
I notice `wfc.releaseAllWorkflowLocks(obj)` in the DeleteFunc. Do we not need that here too? If we do, can we consolidate the shared code into a single method?

Reply:
Also, can we add comments here? (I for one had no idea that the FilterFunc actually purges from the Informer cache, which should have been commented before you arrived I think)

Reply:
I don't think we do, as I'm imagining that FilterFunc returning false then does a Delete, which should call DeleteFunc.

Reply:
True. In that case, why include `wfc.throttler.Remove(key)` in both places then?

Reply:
As long as the workflow didn't run, we shouldn't need to release any locks is my understanding based upon my previous work on the syncManager. @Joibel is correct I believe, but also this means the wfc.throttler.Remove(key) is not needed since the DeleteFunc will get cold